### PR TITLE
pkg/server: add workspace annotation

### PR DESCRIPTION
**Pull request by ibihim**
Signed-off-by: Krzysztof Ostrowski <kostrows@redhat.com>

## Summary

Annotate audit logs with cluster name set in request context.

## Related issue(s)

https://github.com/kcp-dev/kcp/issues/862

Fixes #

### CODE REVIEW

File: pkg/server/handler.go [hunk 1]
```diff
@@ -20,6 +20,8 @@ import (
 	"context"
 	"errors"
 	"fmt"
+	"github.com/emicklei/go-restful"
+	"github.com/kcp-dev/logicalcluster"
```
Comment (sttts) [inline pkg/server/handler.go:24]: order

File: pkg/server/server.go [hunk 2]
```diff
@@ -224,6 +224,7 @@ func (s *Server) Run(ctx context.Context) error {
 		apiHandler = WithClusterScope(apiHandler)
 		apiHandler = WithInClusterServiceAccountRequestRewrite(apiHandler, unsafeServiceAccountPreAuth)
 		apiHandler = WithAcceptHeader(apiHandler)
+		apiHandler = WithClusterAnnotation(apiHandler)
```
Comment (sttts) [inline pkg/server/server.go:227]: now your code runs too early. It's like shells, called from the outside to inside.

**Comment by ibihim** [comment]
Do we have already auditing in KCP?

I created a PR 1042 - add workspace annotation for issue 862 - add logical cluster annotation to audit events, under the assumption, that it exists already. Otherwise we would need to add filters.WithAuditAnnoations and audit options, right?

**Comment by sttts** [comment]
Looks good now. A small e2e would be nice. Like enable the audit file in the generic e2e server flag config, and check that some request arrived with the workspace set.

**Review by sttts — APPROVED** [review]

**Comment by sttts** [comment]
e2e as follow-up.

**Pull request merged**